mempool/clist_mempool.go

```diff
@@ -818,26 +882,73 @@ func (mem *CListMempool) NewIterator() Iterator {
 // Unsafe for concurrent use by multiple goroutines.
 func (iter *CListIterator) WaitNextCh() <-chan Entry {
 	ch := make(chan Entry)
-	// Spawn goroutine that waits for the next entry, saves it locally, and puts it in the channel.
 	go func() {
-		if iter.cursor == nil {
-			// We are at the beginning of the iteration or the saved entry got removed: wait until
-			// the list becomes not empty and select the first entry.
-			<-iter.txs.WaitChan()
-			// Note that Front can return nil.
-			iter.cursor = iter.txs.Front()
-		} else {
-			// Wait for the next entry after the current one.
-			<-iter.cursor.NextWaitChan()
-			// If the current entry is the last one or was removed, Next will return nil.
-			iter.cursor = iter.cursor.Next()
-		}
-		if iter.cursor != nil {
-			ch <- iter.cursor.Value.(Entry)
+		// Add the next entry to the channel if not nil.
+		if entry := iter.Next(); entry != nil {
+			ch <- entry.Value.(Entry)
 		} else {
 			// Unblock the receiver (it will receive nil).
 			close(ch)
 		}
 	}()
 	return ch
 }
+
+// PickLane returns a _valid_ lane on which to iterate, according to the WRR algorithm. A lane is
+// valid if it is not empty or the number of accessed entries in the lane has not yet reached its
+// priority value.
+func (iter *CListIterator) PickLane() types.Lane {
```

Comment:
Is there a task to thoroughly test this function (in particular, that it follows the WRR algorithm)

Reply:
No. It is definitively needed.

Reply:
I am working on it, I pushed some tests yesterday, working on cleaning them up. They caught errors that happened in e2e (where we keep returning the same transactions all the time).